lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -694,6 +695,10 @@ def api_request(method, endpoint, parameters = nil, options = {})
 
       def commit(method, url, parameters = nil, options = {})
         add_expand_parameters(parameters, options) if parameters
+
+        return Response.new(false, 'Invalid API Key provided') if test? && !key(options).start_with?('sk_test')
+        return Response.new(false, 'Invalid API Key provided') if !test? && !key(options).start_with?('sk_live')
```

Comment:
Ah for `response_is_test?` the seems to get it from the response which would correlate with the API key used. But we want to return the error before we make a request so using `test?` is appropriate.

Reply:
gotcha! I guess I'm just wondering why we wouldn't be using `test?` down there too (i.e. was there a reason the boolean wasn't reliable for determining test status), but I might be overthinking it!

Reply:
It looks like it's been like that since it the gateway was created, https://github.com/activemerchant/active_merchant/commit/ca39e94032097c53c2848119431da32ff6da3f31#diff-75f5fee715fd4b5bedbae9c805cbeb103100cdb5c1bf8d56c248aaa31c947e1c. Not sure why they decided to do it that way